File: math/test/LDTrianglListTest.cpp
/**
 * 	
 *  2015/7/2
 *	Imai
 *  
 *  Copyright(c) Live2D Inc. All rights reserved.
 *  [[ CONFIDENTIAL ]]
 */

#include "LDTrianglListTest.h"
#include <QTest>
#include <QDebug>
#include <LDTriangleList.h>
namespace live2d {

LDTriangleList simpleTriangle(){
	LDTriangleList triangles;
	LDTriangle t1(0,1,2);
	triangles.push_back(t1);
	return triangles;	
}


void LDTrianglListTest::findTest_simple()
{
	LDTriangleList triangles=simpleTriangle();	
	LDTriangleList compare=simpleTriangle();
	LDIndexLine line(0,2);
	LDTriangleList resulut=triangles.find(line);
	QCOMPARE(resulut,compare);
	
}

void LDTrianglListTest::toIndexLineListTest_simple()
{
	LDTriangleList triangles=simpleTriangle();

	LDIndexLineList compare;
	LDIndexLine line_1(0,1);
	LDIndexLine line_2(1,2);
	LDIndexLine line_3(2,0);	
	compare.push_back(line_1);
	compare.push_back(line_2);
	compare.push_back(line_3);
	
	LDIndexLineList result= triangles.toIndexLineList();
	
	QCOMPARE(result,compare);
}

void LDTrianglListTest::getOutlinePointIndicesTest_simple()
{
	LDTriangleList triangles=simpleTriangle();
	
	LDPointList points;
	points.push_back(LDPoint(0,0));
	points.push_back(LDPoint(0,100));	
	points.push_back(LDPoint(100,100));
	
	QList<int> compare;
	compare.push_back(0);
	compare.push_back(1);
	compare.push_back(2);
	
	QList<int> result=triangles.getOutlinePointIndices(points);
	
	QCOMPARE(result,compare);	
}

void LDTrianglListTest::getOutlinePointsTest_simple()
{
	LDTriangleList triangles=simpleTriangle();
	LDPointList points;
	points.push_back(LDPoint(0,0));
	points.push_back(LDPoint(0,100));	
	points.push_back(LDPoint(100,100));
	
	LDPointList compare=points;
	
	LDPointList result=triangles.getOutlinePoints(points);

	QCOMPARE(result,compare);	
	
	
}

void LDTrianglListTest::getOutlinePolygonTest_simple()
{
	LDTriangleList triangles=simpleTriangle();
	LDPointList points;
	points.push_back(LDPoint(0,0));
	points.push_back(LDPoint(0,100));	
	points.push_back(LDPoint(100,100));
	
	QPolygonF compare;
	compare.push_back(LDPoint(0,0));
	compare.push_back(LDPoint(0,100));
	compare.push_back(LDPoint(100,100));
	
	QPolygonF result=triangles.getOutlinePolygon(points);
	
	QCOMPARE(result,compare);	
	
}

void LDTrianglListTest::getRelatedTrianglesTest_simple()
{
	LDTriangleList triangles=simpleTriangle();

	LDTriangleList compare=simpleTriangle();
	
	LDTriangleList result = triangles.getRelatedTriangles(0);
	QCOMPARE(result,compare);	
	
}

void LDTrianglListTest::getRelatedLinesTest_simple()
{
	LDTriangleList triangles=simpleTriangle();
	LDIndexLineList compare;
	compare.push_back(LDIndexLine(0,1));
	compare.push_back(LDIndexLine(0,2));
	
	LDIndexLineList result=triangles.getRelatedLines(0);
	QCOMPARE(result,compare);	
	
	
}

void LDTrianglListTest::getRelatedPointIndicesTest_simple()
{
	LDTriangleList triangles=simpleTriangle();
	QList<int> compare;
	compare.push_back(1);
	compare.push_back(2);
	
	QList<int> result=triangles.getRelatedPointIndices(0);
	QCOMPARE(result,compare);	
		
}

} // namespace live2d

File: cs/ConsoleApplication1/LDTrianglListTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace live2d
{
    public class LDTrianglListTest : ITest
    {
        public void test()
        {
            findTest_simple();
            toIndexLineListTest_simple();
            getOutlinePointIndicesTest_simple();
            getOutlinePointsTest_simple();
            getOutlinePolygonTest_simple();
            getRelatedTrianglesTest_simple();
            getRelatedLinesTest_simple();
            getRelatedPointIndicesTest_simple();
        }
        LDTriangleList simpleTriangle()
        {
            LDTriangleList triangles=new LDTriangleList();
            LDTriangle t1=new LDTriangle(0, 1, 2);
            triangles.Add(t1);
            return triangles;
        }


        public void findTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();
            LDTriangleList compare = simpleTriangle();
            LDIndexLine line=new LDIndexLine(0, 2);
            LDTriangleList resulut = triangles.find(line);
            TestUtil.COMPARE(resulut, compare);

        }

        public void toIndexLineListTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();

            LDIndexLineList compare=new LDIndexLineList();
            LDIndexLine line_1=new LDIndexLine(0, 1);
            LDIndexLine line_2=new LDIndexLine(1, 2);
            LDIndexLine line_3 = new LDIndexLine(2, 0);
            compare.Add(line_1);
            compare.Add(line_2);
            compare.Add(line_3);

            LDIndexLineList result = triangles.toIndexLineList();

            TestUtil.COMPARE(result, compare);
        }

        public void getOutlinePointIndicesTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();

            LDPointList points=new LDPointList();
            points.Add(new LDPoint(0, 0));
            points.Add(new LDPoint(0, 100));
            points.Add(new LDPoint(100, 100));

            List<int> compare=new List<int>();
            compare.Add(0);
            compare.Add(1);
            compare.Add(2);

            List<int> result = triangles.getOutlinePointIndices(points);

            TestUtil.COMPARELIST(result, compare);
        }

        public void getOutlinePointsTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();
            LDPointList points=new LDPointList();
            points.Add(new LDPoint(0, 0));
            points.Add(new LDPoint(0, 100));
            points.Add(new LDPoint(100, 100));

            LDPointList compare = points;

            LDPointList result = triangles.getOutlinePoints(points);

            TestUtil.COMPARE(result, compare);


        }

        public void getOutlinePolygonTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();
            LDPointList points=new LDPointList();
            points.Add(new LDPoint(0, 0));
            points.Add(new LDPoint(0, 100));
            points.Add(new LDPoint(100, 100));

            LDPolygon compare=new LDPolygon();
            compare.Add(new LDPoint(0, 0));
            compare.Add(new LDPoint(0, 100));
            compare.Add(new LDPoint(100, 100));

            LDPolygon result = triangles.getOutlinePolygon(points);

            TestUtil.COMPARE(result, compare);

        }

        public void getRelatedTrianglesTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();

            LDTriangleList compare = simpleTriangle();

            LDTriangleList result = triangles.getRelatedTriangles(0);
            TestUtil.COMPARE(result, compare);

        }

        public void getRelatedLinesTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();
            LDIndexLineList compare=new LDIndexLineList();
            compare.Add(new LDIndexLine(0, 1));
            compare.Add(new LDIndexLine(0, 2));

            LDIndexLineList result = triangles.getRelatedLines(0);
            TestUtil.COMPARE(result, compare);


        }

        public void getRelatedPointIndicesTest_simple()
        {
            LDTriangleList triangles = simpleTriangle();
            List<int> compare=new List<int>();
            compare.Add(1);
            compare.Add(2);

            List<int> result = triangles.getRelatedPointIndices(0);
            TestUtil.COMPARELIST(result, compare);

        }

    }
}
